### PR TITLE
Add getting started doc

### PR DESCRIPTION
**Pull request by oshoval**
Signed-off-by: Or Shoval <oshoval@redhat.com>

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+﻿
```
Comment (phoracek) [inline docs/getting-started.md]: Please drop the empty line

File: docs/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+# Kubevirtci
+
+Kubevirtci rule is to create a pre provisioned clusters, publish them as a container image,\
```
Comment (phoracek) [inline docs/getting-started.md]: kubevirtci's role is to create preprovisioned Kubernetes/OpenShift clusters as container images.

File: docs/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+and allow people to run a containerized version of a k8s / open-shift cluster.
+
+Target audience of this project is Developers of kubevirt.
```
Comment (phoracek) [inline docs/getting-started.md]: The target audience should be newcoming developers of kubevirt*ci*.

After reading this document, I should have a basic idea of architecture of kubevirtci providers and be able to create my own one. e.g. a kubernetes provider with IPv6 only.

File: docs/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+```
+git clone https://github.com/kubevirt/kubevirtci.git
+cd kubevirtci
+```
+
+Make sure you have jq installed (can be installed by yum/dnf with the basic repositories).
+
+#### gocli developer guide:
+gocli is a binary that assist in provisioning and spinning (cluster-up) a cluster.
+Uses bazel as build tool.
```
Comment (phoracek) [inline docs/getting-started.md]: I don't think bazel is that important here. Maybe we don't need to mention it at all. Definitely not as one of the first parts about kubevirtci. All I need to use are `make` targets anyway, no?

File: docs/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+Bazel is good mostly for bigger scale projects, where it has declarative dependency tree,\
+which allows to do caching and save lot of build time,\
+also, thanks to gazelle addon, there is a rule for building each target on its own.
+
+#### Install 
+Prerequisites:
```
Comment (phoracek) [inline docs/getting-started.md]: Isn't there a developer doc in the project already teaching how to prepare the dev environment? This doc should reference existing docs while additionally explaining how to create a new provider.

File: docs/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+shutdown vms (so it will be ready to docker commit, stable status).\
+
+container commit (creates an image out of the container) in case all succeed.
+
+# Flow of running make cluster-up
+./cluster-up/cluster/ocp-4.3/provider.sh\
+parse arguments of bash, for example extract pull secret.\
+runs gocli okd run (./cluster-provision/gocli/cmd/okd/run.go according Cobra path)
+
+./cluster-provision/gocli/cmd/okd/run.go\
```
Comment (phoracek) [inline docs/getting-started.md]: Maybe it would be better to use a list instead of enforcing newlines. It would be definitely easier to read.

File: docs/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
+﻿
+# Kubevirtci
```
Comment (phoracek) [inline docs/getting-started.md]: The content of the doc is ok, but the structure is really hard to follow, especially for someone new into project.

Please reference existing docs (like setting up the dev env).

Remember that the target audience are kubevirtci contributors that want to add a new flavour of provider.

The structure should be:

1. What is kubevirtci and how is it helpful, that you already have here
2. What does the running provider consist off
3. What happens during `cluster-up`, one 
4. The structure of the project
5. Simple example of how to create a new provider (e.g. install one additional manifest on k8s), it would be helpful if it was a step by step tutorial from checking out of the code all the way to pushing a PR. It should be enough (or reference enough) for a new dev to contribute.

File: docs/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
+﻿# Kubevirtci
+
+Kubevirtci's role is to create preprovisioned Kubernetes/OpenShift clusters as container images.\
```
Comment (dankenigsberg) [inline docs/getting-started.md]: I'd write
```
The purpose of Kubevirtci is to create pre-provisioned Kubernetes/OpenShift clusters as container images, allowing people to easily run a of a k8s / OpenShift cluster.

Comment (oshoval) [reply docs/getting-started.md]: done

File: docs/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,181 @@
+﻿# Kubevirtci
+
+The purpose of Kubevirtci is to create pre-provisioned Kubernetes/OpenShift clusters as container images,\
```
Comment (dankenigsberg) [inline docs/getting-started.md]: why is the trailing backslash? We should let the HTML rendering decide where to break lines.

File: docs/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+allowing people to easily run a of a k8s / OpenShift cluster.
```
Comment (dankenigsberg) [inline docs/getting-started.md]: s/a of//

File: docs/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+
+The target audience should be newcoming developers of kubevirtci.
```
Comment (dankenigsberg) [inline docs/getting-started.md]: I think that

The target audience *is* people wanting to create a new kubevirtci provider or freshen and existing one.

File: docs/getting-started.md (hunk 9 above)
Comment (oshoval) [reply docs/getting-started.md]: the lines get very long without it, i can remove later and you can see and decide

**Comment by oshoval** [comment]
@RamLavi 
please review,
as we spoke, focusing atm on k8s, after that will add ocp dev guide.
thanks

File: cluster-up/cluster/k8s-1.17.0/dev-guide.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+# Flow of K8s cluster-up (1.17 for example)
+Run
+```
+export KUBEVIRT_PROVIDER=k8s-1.17.0
+make cluster-up
+```
+* Runs ./cluster-up/up.sh which sources the following.
+    * ./cluster-up/cluster/k8s-1.17.0/provider.sh (selected according $KUBEVIRT_PROVIDER)
+    * ./cluster-up/cluster/k8s-provider-common.sh
```
Comment (RamLavi) [inline cluster-up/cluster/k8s-1.17.0/dev-guide.md]: didn't find it in the script. do you mean
source ${KUBEVIRTCI_CLUSTER_PATH}/$KUBEVIRT_PROVIDER/provider.sh
?

Comment (oshoval) [reply cluster-up/cluster/k8s-1.17.0/dev-guide.md]: alias f='find . -name'

[root@modi01 kubevirtci]# f k8s-provider-common.sh
./cluster-up/cluster/k8s-provider-common.sh

cluster-up/cluster/k8s-1.17.0/provider.sh:source ${KUBEVIRTCI_PATH}/cluster/k8s-provider-common.sh
which resolves to hardcoded 
     * ./cluster-up/cluster/k8s-provider-common.sh

Comment (RamLavi) [reply cluster-up/cluster/k8s-1.17.0/dev-guide.md]: ok, but if I am not mistaken, cluster-up/up.sh:11 is 
source ${KUBEVIRTCI_PATH}hack/common.sh
how is it resolved to: ./cluster-up/cluster/k8s-provider-common.sh ?

File: cluster-up/cluster/k8s-1.17.0/dev-guide.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+* Triggers `gocli run` - ./cluster-provision/gocli/cmd/run.go which create the following containers:
```
Comment (RamLavi) [inline cluster-up/cluster/k8s-1.17.0/dev-guide.md]: where the 'gocli run' code in cluster-up/up.sh? 
is it inside one of the bash files? if so, please specify in a hierarchy structure.

File: cluster-up/cluster/k8s-1.17.0/dev-guide.md (hunk 12 above)
Comment (oshoval) [reply cluster-up/cluster/k8s-1.17.0/dev-guide.md]: i meant that 
   * ./cluster-up/cluster/k8s-1.17.0/provider.sh (selected according $KUBEVIRT_PROVIDER)
sources the next line 
   * ./cluster-up/cluster/k8s-provider-common.sh

uploading a chance soon and lets see there whats the status

File: cluster-up/cluster/k8s-1.17.0/dev-guide.md (hunk 13 above)
Comment (oshoval) [reply cluster-up/cluster/k8s-1.17.0/dev-guide.md]: uploading a chance soon and lets see there whats the status

File: cluster-up/cluster/k8s-1.17.0/dev-guide.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
+﻿# kubevirtci K8s provider dev guide.
+
+The purpose of kubevirtci is to create pre-provisioned K8s clusters as container images,
+allowing people to easily run a K8s cluster.
```
[the remark this answers is not in the document]
Comment (oshoval) [reply cluster-up/cluster/k8s-1.17.0/dev-guide.md]: tried

**Comment by oshoval** [comment]
@rmohr 
can you unhold please?
all passed

thanks

**Comment by phoracek** [comment]
/lgtm

Thanks Or! Looks good, very helpful.
